.github/workflows-source/ci-pr-only.yml

```diff
@@ -77,11 +77,17 @@ jobs:
     name: Lock Generate
     <<: *dind-small-setup
     steps:
+      - name: Create GitHub App Token
+        uses: actions/create-github-app-token@v1
+        id: app-token
+        with:
+          app-id: ${{ vars.PR_AUTOMATION_BOT_PUBLIC_APP_ID }}
+          private-key: ${{ secrets.PR_AUTOMATION_BOT_PUBLIC_PRIVATE_KEY }}
       - name: Checkout
         uses: actions/checkout@v4
         with:
           ref: ${{ github.head_ref }}
-          token: ${{ secrets.IDX_PUSH_TO_PR }}
+          token: ${{ steps.app-token.outputs.token }}
```

Comment:
Yes it probably would make sense to unify the approach at some point, but my understanding is that you can either add the token to the checkout step and it will use this GITHUB_TOKEN for the rest of the workflow, or you can explicitly set the GITHUB_TOKEN in the step where it's needed.

Reply:
So because `lock-generate` needs to create a commit, that token is required there, but we could have also passed it into that job specifically.

Reply:
So does that mean the token is needed here? What fails when you remove this line?

Reply:
Ok, I think I understand what you mean. The token is really needed in the `lock-generate` step because it needs to push a commit. 

I wonder if it wouldn't be clearer to move this token to the  `lock-generate` step then...

Reply:
I see I wrote my comments while you were submitting an answer. I understand it now. I would still prefer if the token can be moved to the step that actually needs it. But we could also do that in another PR.